### PR TITLE
Deployment how-to

### PR DESCRIPTION
**Pull request by bemailloux**
It's kind of wordy, but the deployment process is kind of complicated. Let me know if the steps can actually be followed and/or if I'm missing anything.

FYI @Vitalii-Misechko @MykolaGalian @hyehayes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,11 +131,75 @@ npm run test:diff
...
+	getCollectionName
+);
+```
+with
+```js
+return {
+	language: lang,
+	resources: Object.assign(enTranslations.default, translations.default)
+};
+```
```
Comment (NicholasHallman) [inline README.md:181]: Super helpful! great addition

File: README.md [hunk 2]
```diff
@@ -3,62 +3,43 @@
...
+| showOverdueCard | Boolean | true | Shows the overdue assignments card. |
+| showResultsCard | Boolean | true | Shows the results card. |
+| showSystemAccessCard | Boolean | true | Shows the system access card. |
+| showCoursesCol | Boolean | true | Shows the Courses column in the users table. |
+| showDiscussionsCol | Boolean | true | Shows the Discussion activity column in the users table. |
+| showGradeCol | Boolean | true | Shows the Final Grade column in the users table. |
+| showLastAccessCol | Boolean | true | Shows the System Last Access column in the users table. |
+| showTicCol | Boolean | true | Shows the Time in Content column in the users table. |
+| lastAccessThresholdDays | Number | 14 | The number of days since the last system access to show in the system access card. |
+| includeRoles | String | {empty, i.e. all roles} | A list of role ids to include in the dashboard data. Roles that are not in the list will not be shown in the dashboard. |
```
Comment (NicholasHallman) [inline README.md:37]: Awesome! thanks for documenting this

**Review by NicholasHallman — APPROVED** [review]
Very thorough and understandable LGTM

File: README.md [hunk 3]
```diff
@@ -150,11 +131,75 @@ npm run test:diff
...
+#### After dev done (during cert)
+1. Wait for the translations PR to arrive. This PR should target the master branch. It should arrive around 1-1.5 weeks
+   after dev done. Check the [#serge_status](https://d2l.slack.com/archives/CA71JLR4H) channel in Slack for translations
+   status updates.
+
+2. Create a new branch off the r20.xx.xx branch. Cherry-pick the commits from the translations PR to this new branch.
+   Create a second PR with r20.xx.xx as its target.
+
+3. Using the r20.xx.xx-derived branch, verify that the translations don't cause any formatting issues.
+	* NB: When testing on a local LMS, you may need to temporarily disable the OSLO langterms override for the langterms
```
Comment (johngwilkinson) [inline README.md:166]: Why would this be necessary?

**Review by anhill-D2L — APPROVED** [review]

**Review by johngwilkinson — APPROVED** [review]
Thanks for this!

**Comment by johngwilkinson** [comment]
> Visual diff tests failed - pull request #298 has been opened with the updated goldens.

When I was catching up on email, it looked like this was happening a lot, and the natural response was to just keep merging them - which risks missing actual regressions. Hopefully we can fix this flakiness soon.

File: README.md (hunk 3 above)
Comment (bemailloux) [reply README.md:166]: Good question. When I was doing the deployment this time, initially all the terms were still appearing in English despite the new langterms files existing. That's because the OSLO overrides don't automatically get updated just because new langterms files exist, so the original English overrides were overriding the newly translated terms.

Uzair who works on OSLO told me there wasn't an easy way to locally update OSLO terms at the moment. I also tried a couple of things (BSI rebuild, `langterms_only.bat`) to try to update the terms in OSLO, but nothing worked.

**Comment by bemailloux** [comment]
> > Visual diff tests failed - pull request #298 has been opened with the updated goldens.
> 
> When I was catching up on email, it looked like this was happening a lot, and the natural response was to just keep merging them - which risks missing actual regressions. Hopefully we can fix this flakiness soon.

@NicholasHallman clarified the correct course of action here: https://github.com/Brightspace/insights-engagement-dashboard/pull/267#issuecomment-752727901

To stop the flakes we could try increasing the test tolerances from 0 to 0.05, like @Vitalii-Misechko did here: https://github.com/Brightspace/insights-engagement-dashboard/blob/4892db14eb3cfd25a532fd01fd80b28fb50e6bd6/test/visual-diff/engagement-dashboard-skeleton.visual-diff.js#L7. I don't know for sure if it would always work though, as (if I'm understanding correctly) tolerance looks at the difference between the colour of two pixels.

**Comment by Vitalii-Misechko** [comment]
> > > Visual diff tests failed - pull request #298 has been opened with the updated goldens.
> > 
> > 
> > When I was catching up on email, it looked like this was happening a lot, and the natural response was to just keep merging them - which risks missing actual regressions. Hopefully we can fix this flakiness soon.
> 
> @NicholasHallman clarified the correct course of action here: [#267 (comment)](https://github.com/Brightspace/insights-engagement-dashboard/pull/267#issuecomment-752727901)
> 
> TL;DR if it's not an actual difference, just re-run the test instead of merging
> 
> To stop the flakes we could try increasing the test tolerances from 0 to 0.05, like @Vitalii-Misechko did here:
> 
> https://github.com/Brightspace/insights-engagement-dashboard/blob/4892db14eb3cfd25a532fd01fd80b28fb50e6bd6/test/visual-diff/engagement-dashboard-skeleton.visual-diff.js#L7
> 
> . I don't know for sure if it would always work though, as (if I'm understanding correctly) tolerance looks at the difference between the colour of two pixels.

I'm not sure if setting the tolerance setting will help us. Worth trying though. I remember a case when I faced a lot of visual diff PRs with only screenshot size changes (screenshot size is not affected by the tolerance setting). It looks to me that we might want to run visual diffs manually and approve as a step of quality notes to avoid a lot of messages from visual diffs automation.
